crates/rpc/src/handlers/validator.rs

```diff
@@ -192,3 +203,60 @@ pub async fn post_validators_from_state(
 
     get_validators_from_state(state_id, id_query, status_query, db).await
 }
+pub async fn get_validator_balances_from_state(
+    state_id: ID,
+    query: ValidatorBalanceQuery,
+    db: ReamDB,
+) -> Result<impl Reply, Rejection> {
+    let state = get_state_from_id(state_id, &db).await?;
+
+    if let Some(ref ids) = query.id {
+        if ids.len() > 1000 {
+            return Err(ApiError::TooManyValidatorsIds())?;
+        }
+    }
+
+    let filter: Option<HashSet<PubKey>> = match &query.id {
+        Some(ids) if ids.is_empty() => None,
+        Some(ids) => {
+            let addresses: HashSet<PubKey> = ids
+                .iter()
+                .filter_map(|v| match v {
+                    ValidatorID::Address(addr) => Some(addr.clone()),
+                    _ => None,
+                })
+                .collect();
+            Some(addresses)
+        }
+        None => None,
+    };
```

Comment:
aren't we missing the `validator index` case?

Reply:
yeah got it fixed, mis-read the api requirements from the reference before